### ISSUE TITLE
Possibility for specifying other things to collect in the path, beside the key.

### ISSUE THREAD

**Issue by franciscolourenco** [bojand/prop-search#2]
For example, we might be interested in receiving an path with the the values, instead of the keys. 
In my case I'm interested in a specific property of each object in the path.

In the same way it is possible to specify a search function, it would be nice if it was possible to specify a function which gets the value which gets added to the path array.

**Comment by bojand** [comment]
Hello, thanks for the comment. Can you elaborate some more on what you mean exactly and what type of functionality you have in mind? Once we have the search results the actual result value is easy to retrieve using [lodash.get()](https://lodash.com/docs/4.17.4#get), [mpath](https://github.com/aheckmann/mpath), or [dot-prop](https://github.com/sindresorhus/dot-prop). A more versatile object traversal / walk utility is [js-traverse](https://github.com/substack/js-traverse) which may be more suitable for your use case and scenario? But again not sure exactly what you mean so more details would be helpful. Thanks.

**Comment by franciscolourenco** [comment]
@bojand yes, that is what I meant, and I'm aware I can just `path.map` to retrieve the values I want. However it would be more efficient if we got the desired values as we traverse the tree.
